File: devlibrary/src/main/java/com/example/devlibrary/helper/RealmHelper.java
package com.example.devlibrary.helper;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class RealmHelper {
    private static final String DATABASE_NAME = "WanAndroid.realm";

    public static Realm newRealmInstance(){
        return Realm.getInstance(new RealmConfiguration.Builder()
                .name(DATABASE_NAME)
                .deleteRealmIfMigrationNeeded()
                .build());
    }
}

File: baselib/src/main/java/com/wan/baselib/helper/RealmHelper.kt
package com.wan.baselib.helper

import io.realm.Realm
import io.realm.RealmConfiguration

object RealmHelper {

    private const val DATABASE_NAME = "WanAndroid.realm"

    val realm: Realm
        get() = newRealmInstance()

    private fun newRealmInstance(): Realm {
        return Realm.getInstance(
            RealmConfiguration.Builder()
                .name(DATABASE_NAME)
                .schemaVersion(0)
                .deleteRealmIfMigrationNeeded()
                .build()
        )
    }
}